tests/run_test.go

```diff
@@ -0,0 +1,217 @@
+
+	_, err := os.Stat(filepath.Join(pebbleDir, "svc1"))
+	if err == nil {
+		t.Error("pebble run --hold failed, services are still started")
+	} else {
+		if !os.IsNotExist(err) {
+			t.Errorf("Error checking service %s: %v", "svc1", err)
+			fmt.Printf("Error checking the file: %v\n", err)
+		}
+	}
+}
+
+func TestHttpPort(t *testing.T) {
+	pebbleDir := t.TempDir()
+
+	port := "4000"
+	logsCh := pebbleRun(t, pebbleDir, "--http=:"+port)
+	expectedLogs := []string{"Started daemon"}
+	if err := waitForLogs(logsCh, expectedLogs, time.Second*3); err != nil {
+		t.Errorf("Error waiting for logs: %v", err)
+	}
+
+	if !isPortUsedByProcess(t, port, "pebble") {
+		t.Errorf("Pebble is not listening on port %s", port)
+	}
+}
+
+func TestVerbose(t *testing.T) {
+	pebbleDir := t.TempDir()
+
+	layerYAML := `
+services:
+    svc1:
+        override: replace
+        command: /bin/sh -c "{{.svc1Cmd}}"
+        startup: enabled
+`
+	layersFileName := "001-simple-layer.yaml"
+	svc1Cmd := fmt.Sprintf("cat %s; sleep 1000", filepath.Join(pebbleDir, "layers", layersFileName))
+	layerYAML = strings.Replace(layerYAML, "{{.svc1Cmd}}", svc1Cmd, -1)
+
+	createLayer(t, pebbleDir, layersFileName, layerYAML)
+
+	logsCh := pebbleRun(t, pebbleDir, "--verbose")
+	expectedLogs := []string{
+		"Started daemon",
+		"services:",
+		"svc1:",
+		"override: replace",
+		"startup: enabled",
+		"command: /bin/sh -c",
+	}
+	if err := waitForLogs(logsCh, expectedLogs, time.Second*3); err != nil {
+		t.Errorf("Error waiting for logs: %v", err)
+	}
+}
+
+func TestArgs(t *testing.T) {
+	pebbleDir := t.TempDir()
+
+	layerYAML := `
+services:
+    svc1:
+        override: replace
+        command: /bin/sh
+        startup: enabled
+`
+	layersFileName := "001-simple-layer.yaml"
+	svc1Cmd := fmt.Sprintf("cat %s; sleep 1000", filepath.Join(pebbleDir, "layers", layersFileName))
+	layerYAML = strings.Replace(layerYAML, "{{.svc1Cmd}}", svc1Cmd, -1)
+
+	createLayer(t, pebbleDir, layersFileName, layerYAML)
+
+	logsCh := pebbleRun(t, pebbleDir, "--verbose",
+		"--args",
+		"svc1",
+		"-c",
+		fmt.Sprintf("cat %s; sleep 1000", filepath.Join(pebbleDir, "layers", layersFileName)),
+	)
+	expectedLogs := []string{
+		"Started daemon",
+		"services:",
+		"svc1:",
+		"override: replace",
+		"startup: enabled",
+		"command: /bin/sh",
+	}
+	if err := waitForLogs(logsCh, expectedLogs, time.Second*3); err != nil {
+		t.Errorf("Error waiting for logs: %v", err)
+	}
+}
+
+func TestIdentities(t *testing.T) {
+	pebbleDir := t.TempDir()
+
+	identitiesYAML := `
+identities:
+    bob:
+        access: admin
+        local:
+            user-id: 42
+    alice:
+        access: read
+        local:
+            user-id: 2000
+`
+	identitiesFileName := "idents-add.yaml"
+	createIdentitiesFile(t, pebbleDir, identitiesFileName, identitiesYAML)
+
+	logsCh := pebbleRun(t, pebbleDir, "--identities="+filepath.Join(pebbleDir, identitiesFileName))
+	expectedLogs := []string{
+		"Started daemon",
+		"POST /v1/services",
```

Comment:
I don't think we should bother checking the logs in the tests (I don't think stable log formatting should even be part of the contract). So let's just check the expected behaviour -- the output of `pebble identity` below.

Reply:
OK, then I will change it to the same as `TestHold` where after `pebble run` we sleep 100 millisecond just to be sure the daemon is up and running before issuing `pebble identity` commands.